tools/webhook-apicoverage/resourcetree/node.go

```diff
@@ -0,0 +1,25 @@
+package resourcetree
+
+import "reflect"
+
+//node.go contains types and interfaces pertaining to nodes inside resource tree.
+
+//INode interface defines methods that can be performed on each node in the resource tree.
+type INode interface {
+	data() nodeData
+	initialize(field string, parent INode, t reflect.Type, tr *ResourceTree)
+}
+
+//nodeData is the data stored in each node of the resource tree.
+type nodeData struct {
+	field string // Represents the Name of the field e.g. field name inside the struct.
+	tree *ResourceTree // Reference back to the resource tree.
+	kind reflect.Kind
+	fieldType reflect.Type
+	nodePath string // Path in the resource tree reaching this node.
+	pkg string // Package name where the type of the node(FieldType) belongs.
+	parent INode // Link back to parent.
+	children map[string]INode // Child nodes are keyed using field names(Field).
+	leafNode bool
```

Comment:
nodeData contains quite a few data such as tree, leafNode, pkg that can be computed at runtime. I assume that you've done compute vs. storage tradeoff consideration. can you pls add comment on why you decide to stick with persist the data that can be computed at runtime?

Reply:
I have added comments for fields like tree and leafNode. I have also removed a couple of them like pkg and Kind. Thanks for this, I got a chance to re-evaluate all these fields.